cmd/hipsterfy/Hipsterfy/Server.hs

```diff
@@ -74,20 +74,19 @@ handleCompare = post "/compare" $ do
   -- Check that the friend code is valid.
   friendCode <- param "friend-code"
   maybeFriend <- getUserByFriendCode friendCode
-  friend <- case maybeFriend of
-    Just f -> return f
-    -- TODO: display an error message for invalid friend codes.
-    Nothing -> redirect "/"
-
-  -- Load followed artists.
-  -- TODO: load these in parallel
-  yourArtists <- getFollowedArtists $ userID user
-  friendArtists <- getFollowedArtists $ userID friend
+  case maybeFriend of
+    Just f -> do 
```

Comment:
[nit] Since `friend` is no longer a used name, I'd expand `f` to `friend` for readability.